node/network/availability-distribution/src/lib.rs

```diff
@@ -14,1228 +14,116 @@
-			));
-
-			metrics.on_chunk_distributed();
-		}
-	}
-}
-
-async fn send_batch_to_network(
-	ctx: &mut impl SubsystemContext,
-	batch: Vec<(Vec<PeerId>, protocol_v1::ValidationProtocol)>,
-) {
-	if !batch.is_empty() {
-		ctx.send_message(NetworkBridgeMessage::SendValidationMessages(batch).into()).await
-	}
-}
-
-// Send the difference between two views which were not sent
-// to that particular peer.
-#[tracing::instrument(level = "trace", skip(ctx, metrics), fields(subsystem = LOG_TARGET))]
-async fn handle_peer_view_change<Context>(
-	ctx: &mut Context,
-	state: &mut ProtocolState,
-	origin: PeerId,
-	view: View,
-	metrics: &Metrics,
-)
-where
-	Context: SubsystemContext<Message = AvailabilityDistributionMessage>,
-{
-	let current = state.peer_views.entry(origin.clone()).or_default();
-
-	let added: Vec<Hash> = view.difference(&*current).cloned().collect();
-
-	*current = view;
-
-	if added.is_empty() {
-		return
-	}
-
-	// only contains the intersection of what we are interested and
-	// the union of all relay parent's candidates.
-	let added_candidates = state.cached_live_candidates_unioned(added.iter());
-
-	// Send all messages we've seen before and the peer is now interested in.
-	let mut batch = Vec::new();
-	for candidate_hash in added_candidates {
-		let per_candidate = match state.per_candidate.get_mut(&candidate_hash) {
-			Some(p) => p,
-			None => continue,
-		};
-
-		// obtain the relevant chunk indices not sent yet
-		let messages = ((0 as ValidatorIndex)..(per_candidate.validators.len() as ValidatorIndex))
-			.into_iter()
-			.filter_map(|erasure_chunk_index: ValidatorIndex| {
-				// try to pick up the message from the message vault
-				// so we send as much as we have
-				per_candidate
-					.message_vault
-					.get(&erasure_chunk_index)
-					.filter(|_| per_candidate.message_required_by_peer(&origin, erasure_chunk_index))
-			})
-			.cloned()
-			.collect::<HashSet<_>>();
-
-		add_tracked_messages_to_batch(&mut batch, per_candidate, metrics, vec![origin.clone()], messages);
-	}
+const LOG_TARGET: &'static str = "availability_distribution";
 
-	send_batch_to_network(ctx, batch).await;
-}
+/// Availability Distribution metrics.
+type Metrics = ();
 
-/// Obtain the first key which has a signing key.
-/// Returns the index within the validator set as `ValidatorIndex`, if there exists one,
-/// otherwise, `None` is returned.
-async fn obtain_our_validator_index(
-	validators: &[ValidatorId],
+/// The availability distribution subsystem.
+pub struct AvailabilityDistributionSubsystem {
+	/// Pointer to a keystore, which is required for determining this nodes validator index.
 	keystore: SyncCryptoStorePtr,
-) -> Option<ValidatorIndex> {
-	for (idx, validator) in validators.iter().enumerate() {
-		if CryptoStore::has_keys(
-			&*keystore,
-			&[(validator.to_raw_vec(), PARACHAIN_KEY_TYPE_ID)],
-		)
-		.await
-		{
-			return Some(idx as ValidatorIndex);
-		}
-	}
-	None
+	//// Prometheus metrics.
+	// metrics: Metrics,
```

Comment:
This is commented out?

Reply:
Already working on a follow up, adding it back in.